### PR TITLE
egui_extras: feature `svg_text`

### PR DESCRIPTION
**Pull request by xNWP**
# Added
- Adds svg text rendering support via the `svg_text` feature flag.

# Changed
- Updates resvg from `0.37` to `0.42`.

### CODE REVIEW

File: crates/egui_extras/src/image.rs [hunk 1]
```diff
@@ -235,45 +235,49 @@ pub fn load_svg_bytes_with_size(
...
-            size = size
-                .scale_to_height(h)
-                .ok_or_else(|| format!("Failed to scale SVG to height {h}"))?;
-        }
-        Some(SizeHint::Width(w)) => {
-            size = size
-                .scale_to_width(w)
-                .ok_or_else(|| format!("Failed to scale SVG to width {w}"))?;
-        }
+    #[allow(unused_mut)]
```
Comment (murl-digital) [inline crates/egui_extras/src/image.rs:242]: is this here because clippy gets confused?

**Review by murl-digital — COMMENTED** [review]
that allow(unused_mut) caught my eye but otherwise lgtm

**Comment by xNWP** [comment]
> that allow(unused_mut) caught my eye but otherwise lgtm
> 
> edit: i should really read commit messages first, could you put in a comment explaining it?

commented!

**Comment by murl-digital** [comment]
i think for demonstration purposes (and to make testing easier in the future), it'd be a good idea to add an example that shows text in svgs in action

**Review by emilk — APPROVED** [review]
I agree that an example would be nice, but LGTM anyway.

Would this text rendering work on web?

**Comment by xNWP** [comment]
> Would this text rendering work on web?

I was wondering the same thing shortly after making the PR 😅, I'm not great at web dev stuff but I'll take a look at it now and test some things.

> i think for demonstration purposes (and to make testing easier in the future), it'd be a good idea to add an example that shows text in svgs in action

after checking the web stuff I'll also look at how y'all test code and see what I can do 😊

**Comment by xNWP** [comment]
> Would this text rendering work on web?

Short answer, no.
Long answer is that the main workhorse of this PR is the `load_system_fonts` function on the Font Database type that `resvg` re-exports from the `fontdb` crate (https://github.com/RazrFalcon/fontdb/blob/fd24f8c9d7b57f30a6f6ba560c697a7854c94858/src/lib.rs#L352). This function is just a conditional compilation switch based on the target os which simply does nothing for wasm. I looked into how support could be added to this crate, however it looks like the way to do it would be the Local Font Access API which has little to no browser support (only experimental Chrome + Edge support) https://developer.mozilla.org/en-US/docs/Web/API/Local_Font_Access_API

**Comment by emilk** [comment]
Thanks for testing web. Please document your findings, i.e. document that the `svg_text` won't work on web, and why (linking to relevant issues, if any).

We should also figure out a solution to the duplicated dependencies.

**Comment by xNWP** [comment]
Disregard this PR (for now).

Apologies for the delay in this, adhd is a bitch. But while looking at this again I realized that I mistakenly have it set up in a way that the system fonts are being loaded EVERY single egui loads a new svg, instead it should be loaded once (during the install image loaders call likely), and that font database should be reused by the SVG loader. I also realized that I could likely solve the problem of no fonts on the web by allowing you to load/embed the font data directly (moving the problem onto the egui user). When I find the time to reimplement this in a better way I'll open a new PR and reference this one :)

**Pull request closed without merging**